commcare_connect/form_receiver/processor.py

```diff
@@ -151,6 +152,14 @@ def process_deliver_unit(user, xform: XForm, app: CommCareApp, opportunity: Oppo
         )
     )
     claim = OpportunityClaim.objects.get(opportunity_access__opportunity=opportunity, opportunity_access__user=user)
+    entity_id = deliver_unit_block.get("entity_id")
+    entity_name = deliver_unit_block.get("entity_name")
+    completed_work, _ = CompletedWork.objects.get_or_create(
+        opportunity=opportunity,
+        entity_id=entity_id,
+        entity_name=entity_name,
+        payment_unit=deliver_unit.payment_unit,
```

Comment:
do deliver units have payment units?

Reply:
Yup, deliver units have a foreign key to a payment unit